hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -193,6 +193,31 @@ func (r *HostedClusterReconciler) SetupWithManager(mgr ctrl.Manager, createOrUpd
 	return builder.Complete(r)
 }
 
+// managedResources are all the resources that are managed as childresources for a HostedCluster
+func managedResources() []client.Object {
+	return []client.Object{
+		&capiawsv1.AWSCluster{},
+		&hyperv1.HostedControlPlane{},
+		&capiv1.Cluster{},
+		&appsv1.Deployment{},
+		&prometheusoperatorv1.PodMonitor{},
+		&networkingv1.NetworkPolicy{},
+		&rbacv1.ClusterRole{},
+		&rbacv1.ClusterRoleBinding{},
+		&rbacv1.Role{},
+		&rbacv1.RoleBinding{},
+		&corev1.ConfigMap{},
+		&corev1.Secret{},
+		&corev1.Namespace{},
+		&corev1.ServiceAccount{},
+		&corev1.Service{},
+		&routev1.Route{},
+		&agentv1.AgentCluster{},
+		&capiibmv1.IBMVPCCluster{},
+		&capikubevirt.KubevirtCluster{},
```

Comment:
Just a tangential discussion... reminds me of our brief talk about DAGs. There is within the project some kind of implicit graph rooted at the HostedCluster. This list exposes part of it in a way, as do the manifest functions. e.g. it seems like every one of the items in this list has a corresponding manifest function (that encodes both the type and the name). The relationships themselves are not encoded statically anywhere though (and there could be indirection between the HostedCluster and some of these)...